### DISCUSSION TITLE
What are "testtools" that I should use instead of unitttest? :-)

### DISCUSSION THREAD

**Discussion by liborjelinek** [falconry/falcon#2155] [General]
On https://falcon.readthedocs.io/en/stable/api/testing.html#other-helpers:
![image](https://github.com/falconry/falcon/assets/383059/72334574-b99d-4f70-a9c2-d23006e5ed9d)

**Comment by vytas7** [comment]
Hi @mattwarrick,
I don't think you should necessarily use `testtools` instead of `unittest`; the documentation just means if `falcon.testing` finds `testtools`, then [`TestCase`](https://falcon.readthedocs.io/en/stable/api/testing.html#test-cases) is instead based upon `testtools.TestCase`. Furthermore, if you use the popular [`pytest`](https://docs.pytest.org/), you don't even need to inherit from `TestCase` at all, you can simply write test methods and/or classes according to [`pytest`'s usage patterns](https://falcon.readthedocs.io/en/stable/user/tutorial.html#testing-your-application).

Personally, I never use `testtools`, I just either use `pytest` directly, or write `unittest.TestCase`-based tests but often opt to use `pytest` as a runner for these. Falcon's integration with `testtools` precedes my history of being a Falcon user and maintainer :smiling_imp: Maybe @kgriffs can share the story behind it :slightly_smiling_face:
It seems, however, that [`testtools` are still maintained](https://pypi.org/project/testtools/) and used, so maybe it doesn't hurt for Falcon to keep this integration? In any case, I've created an issue to track this: #2156.

**Comment by kgriffs** [comment]
This was originally done to support some OpenStack projects since testtools was in common use in that community way back when.
